baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/DangerousCompletableFutureUsage.java

```diff
@@ -0,0 +1,103 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Type;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "DangerousCompletableFutureUsage",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.ERROR,
+        summary = "Disallow CompletableFuture asynchronous operations without an Executor.")
+public final class DangerousCompletableFutureUsage
+        extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+    private static final long serialVersionUID = 1L;
+
+    private static final String ERROR_MESSAGE = "Should not use CompletableFuture methods without specifying a "
+            + "custom executor service. Doing so has very subtle and potentially severe performance implications, so "
+            + "in you're better off using your own executor service which allows you to provide instrumentation "
+            + "and specify the desired parallelism (i.e. the max number of concurrent tasks that will be submitted).\n"
+            + "By default, CompletableFuture uses the globally shared ForkJoinPool. Fork/join pools implement "
+            + "work-stealing, where any thread might steal a task from a different thread's queue when blocked "
+            + "waiting for a subtask to complete. This might not seem like an issue at first glance, but if you rely"
+            + "on the ForkJoinPool for short tasks extensively throughout your codebase and later on you add one "
+            + "piece of code that uses the pool for long (e.g. I/O) tasks, the other parts of your codebase that"
+            + "you'd expect to have consistent performance might experience performance degradation for no apparent "
+            + "reason.\n"
+            + "If you're absolutely certain that the ForkJoinPool is correct, please pass ForkJoinPool.commonPool()"
+            + "directly, ideally with a comment explaining why it is ideal.";
```

Comment:
Slightly modified message from `DangerousParallelStreamUsage.ERROR_MESSAGE`

Reply:
This is great, love the explanation.